integration/proxy/proxy_helpers.go

```diff
@@ -604,26 +627,33 @@ func (m *mockAWSALBProxy) serve(ctx context.Context, t *testing.T) {
 
 		conn, err := m.Accept()
 		if err != nil {
-			if utils.IsOKNetworkError(err) {
-				return
-			}
-			require.NoError(t, err)
+			logrus.WithError(err).Debugf("Failed to accept conn.")
```

Comment:
Suggestion: Handle the errors from the goroutine and assert them. This will make easier to identify why the test failed.

Here's a small example using a channel and `Cleanup`: https://go.dev/play/p/vsGPhW7Oc92 (this way we have both errors, from the test and from the goroutine).

Reply:
Thanks for the suggestion! I was trying it out but then realized we don't know how many errors may come in to create a buffered chan. Will just keep it simple for now as it is.